### PR TITLE
[SPARK-44522][BUILD] Upgrade `scala-xml` to 2.2.0

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to upgrade `scala-xml` from 2.1.0 to 2.2.0.

### Why are the changes needed?
The new version bring some bug fix like:
- https://github.com/scala/scala-xml/pull/651
- https://github.com/scala/scala-xml/pull/677

The full release notes as follows:
- https://github.com/scala/scala-xml/releases/tag/v2.2.0

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass GitHub Actions
- Checked Scala 2.13, all Scala test passed: https://github.com/LuciferYang/spark/runs/15278359785

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -205,6 +205,7 @@ jobs:
       HIVE_PROFILE: ${{ matrix.hive }}
       GITHUB_PREV_SHA: ${{ github.event.before }}
       SPARK_LOCAL_IP: localhost
+      SCALA_PROFILE: scala2.13
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:208]: test scala 2.13, will revert after test pass

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs). Thank you, @LuciferYang .

**Comment by LuciferYang** [comment]
For scala 2.13, all scala test passed. 

https://github.com/LuciferYang/spark/runs/15278359785

`pyspark-pandas-slow-connect` failure is well known

```
Building wheels for collected packages: pyspark
  Building wheel for pyspark (setup.py): started
  Building wheel for pyspark (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [1724 lines of output]
      /tmp/tmp.Ex2gmKU9bV/3.9/lib/python3.9/site-packages/setuptools/dist.py:755: SetuptoolsDeprecationWarning: Invalid dash-separated options
      !!
      
              ********************************************************************************
              Usage of dash-separated 'description-file' will not be supported in future
              versions. Please use the underscore name 'description_file' instead.
      
Error: No space left on device : '/home/runner/runners/2.306.0/_diag/pages/9b7af01d-9255-4597-a103-2b1fbff3f418_551a9d66-8149-5c0e-5e40-9b55ab3d9173_1.log'
```

**Comment by LuciferYang** [comment]
> For scala 2.13, all scala test passed.
> 
> https://github.com/LuciferYang/spark/runs/15278359785
> 
> `pyspark-pandas-slow-connect` failure is well known
> 
> ```
> Building wheels for collected packages: pyspark
>   Building wheel for pyspark (setup.py): started
>   Building wheel for pyspark (setup.py): finished with status 'error'
>   error: subprocess-exited-with-error
>   
>   × python setup.py bdist_wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [1724 lines of output]
>       /tmp/tmp.Ex2gmKU9bV/3.9/lib/python3.9/site-packages/setuptools/dist.py:755: SetuptoolsDeprecationWarning: Invalid dash-separated options
>       !!
>       
>               ********************************************************************************
>               Usage of dash-separated 'description-file' will not be supported in future
>               versions. Please use the underscore name 'description_file' instead.
>       
> Error: No space left on device : '/home/runner/runners/2.306.0/_diag/pages/9b7af01d-9255-4597-a103-2b1fbff3f418_551a9d66-8149-5c0e-5e40-9b55ab3d9173_1.log'
> ```

Seems https://github.com/apache/spark/pull/42115/files try fix this

**Comment by srowen** [comment]
Looks OK just re-run tests

**Comment by LuciferYang** [comment]
rebase

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
Thanks @srowen

**Pull request closed without merging**